README.md

```diff
@@ -1,23 +1,39 @@
 # just-the-docs-template
 
-This is a *bare-minimum* template to create a [Jekyll] site that:
+## Skip the details, get me going!
 
-- uses the [Just the Docs] theme;
-- can be built and published on [GitHub Pages];
-- can be built and previewed locally, and published on other platforms.
+<div align="center">
 
-More specifically, the created site:
+[![Open in Dev Containers](https://img.shields.io/static/v1?label=Dev%20Container&message=Open&color=blue&logo=visualstudiocode&style=for-the-badge)](https://vscode.dev/redirect?url=vscode://ms-vscode-remote.remote-containers/cloneInVolume?url=https://github.com/just-the-docs/just-the-docs-template)
+[![Open in Codespaces](https://img.shields.io/static/v1?label=CodeSpaces&message=Launch&color=blue&logo=github&style=for-the-badge)](https://github.com/codespaces/new/just-the-docs/just-the-docs-template)
 
-- uses a gem-based approach, i.e. uses a `Gemfile` and loads the `just-the-docs` gem;
-- uses the [GitHub Pages / Actions workflow] to build and publish the site on GitHub Pages.
+</div>
 
-To get started with creating a site, just click "[use this template]"!
+The **DevContainer** requires VSCode and Docker to be installed on your machine.
 
-If you want to maintain your docs in the `docs` directory of an existing project repo, see [Hosting your docs from an existing project repo](#hosting-your-docs-from-an-existing-project-repo).
+The **CodeSpace** requires nothing of that and is the ideal solution for beginners.
+
+You will be asked to create a new repository when you push your first commit. <br />
+(You can also create a new repository beforehand, by clicking the green "Use this template"-Button)
```

Comment:
I'm a bit opinionated here, but I don't want to suggest dev containers (or codespaces) as the "first" option in our README; to me, the lowest-friction setup is the "use a template button". Partially this is because I suspect many of our users don't use VSCode and/or have Docker installed on their system (particularly non-technical folks), and Codespaces is a different interface than normal.

I'm open to suggestions here though (especially Codespaces, since I think it actually is really low-friction if it works well!). What are your thoughts?

Reply:
My idea behind these changes is to provide a clear and simple start for inexperienced users. Sometimes clear and simple requires taking decisions away from the users, at least initially. 

Instead of "you can do x or you can do y, but then you need to take care about z" it's more like "just click here and everything will be alright". We can still offer these options, but in my view they belong in our regular documentation in the install section.

Second, I want to provide an easy way to try our template without having to setup many dependencies or reading the install instructions.

I can't tell how many users are using which editor - just based on the numbers ([here](https://survey.stackoverflow.co/2023/#other-tools) and [here](https://survey.stackoverflow.co/2023/#integrated-development-environment) I'd say providing ease-of-access-functions for the most common tool isn't bad per se. And I'd even argue that 18.8 mil downloads for the devcontainers extension in vscode is a very huge number compared to any extension. Codespaces btw uses the same vscode interface than the desktop apps, or vscode.dev, or github.dev.

I'm targeting inexperienced users here. There's a high chance they have vscode installed. Everything else will be taken care of by vscode, even installing docker if I'm not mistaken. Everyone else can checkout all possible options in our install instructions in the main documentation.

You mentioned the "use this template"-button. I thought about leaving it in there as first option. I even tested it for myself. In the end I think it's more confusing: Click the button in repository A, select a name for the copy. Then go to the copy and click one of the two buttons there. Being honest I need to admit github and microsoft did a pretty good job hiding that process in vscode. Launching a devcontainer or a codespace from a template repository looks like doing the same with any other repository - except they don't checkout the .git-directory. As soon as the user tries to commit his changes, he will be asked where to store them, with direct options for creating a new github repo in his namespace. From my point of view this is a pretty neat user experience.

Reply:
Thanks for explaining your reasoning! I generally agree with the idea, but I'm not a huge fan of the framing of the text. Essentially, I *don't* want users to "skip the details" that explain how the website is built and deployed, since this is both:

- pretty essential to the use of the theme
- (IMO) an active choice, which Codespaces makes *for you* without telling you.
    - I think users should *actively* click a button that forks the repository, rather than having a tool do it in the background; this gives them more visibility into what's happening.

This is mostly a request for minor tweaking of the README.

---

Separately,

> I'm targeting inexperienced users here. There's a high chance they have vscode installed. Everything else will be taken care of by vscode, even installing docker if I'm not mistaken. Everyone else can checkout all possible options in our install instructions in the main documentation.

Hm, I don't think this is true.

- **most importantly**: [users need to install Docker themselves to use Dev Containers](https://code.visualstudio.com/docs/devcontainers/containers). I don't think it's reasonable for our preferred template to require users to install Docker; Docker is nontrivially hard to install, and it's not *as useful* for users in this case (editing a static site generator, which is an isolated micro-app) compared to other use-cases where containerized applications are more common.
- anecdotally, quite a few of the users I respond to on GH issues are using a terminal-based text editor, and in the teaching that I do I still see lots of variance in code editors

More broadly, I don't think we (as theme developers) should have strong opinions on anything more than:

1. install version `x` of Ruby
2. make sure to use at least version `y` of Jekyll